src/main/java/org/reactome/release/cosmicupdate/Main.java

```diff
@@ -0,0 +1,337 @@
+		{
+			logger.error("Error with SQL Connection", e);
+			e.printStackTrace();
+		} catch (Exception e) {
+			logger.error("Exception caught", e);
+		}
+		logger.info("All done.");
+		
+	}
+
+	private Properties getConfig()
+	{
+		boolean noConfigSpecified = this.configPath != null && this.configPath.trim().isEmpty();
+//		boolean noAuthConfigSpecified = Main.authPath != null && !Main.authPath.trim().isEmpty();
+		if (!noConfigSpecified /* && !noAuthConfigSpecified */)
+		{
+			try(Reader configReader = new FileReader(this.configPath);
+			/* Reader authReader = new FileReader(Main.authPath) */)
+			{
+				Properties configProps = new Properties();
+				configProps.load(configReader);
+				
+//				Properties authProps = new Properties();
+//				authProps.load(authReader);
+
+				// These default to files in the same directory where the program is running.
+				Main.COSMICMutantExport = configProps.getProperty("pathToMutantExportFile", "./CosmicMutantExport.tsv");
+				Main.COSMICFusionExport = configProps.getProperty("pathToFusionExportFile", "./CosmicFusionExport.tsv");
+				Main.COSMICMutationTracking = configProps.getProperty("pathToMutationTrackingFile", "./CosmicMutationTracking.tsv");
+				// These have NO default.
+				Main.COSMICUsername = configProps.getProperty("cosmic.user");
+				Main.COSMICPassword = configProps.getProperty("cosmic.password");
+				Main.personId = Long.parseLong(configProps.getProperty("personId"));
+				// These must be set in the properties file.
+				Main.COSMICMutantExportURL = configProps.getProperty("urlToMutantExportFile");
+				Main.COSMICFusionExportURL = configProps.getProperty("urlToFusionExportFile");
+				Main.COSMICMutationTrackingURL = configProps.getProperty("urlToMutationTrackingFile");
+//				Main cosmicUpdateStep = new Main();
+//				JCommander.newBuilder().addObject(cosmicUpdateStep).build().parse(args);
+//				cosmicUpdateStep.executeStep(configProps);
+				return configProps;
+			}
+			catch (FileNotFoundException e)
+			{
+				logger.fatal("FileNotFoundException occurred while trying to load the config file \""+this.configPath+"\"", e);
+			}
+			catch (IOException e)
+			{
+				logger.fatal("IOException occurred while trying to load the config file.", e);
+			}
+		}
+		else
+		{
+			if (noConfigSpecified)
+			{
+				logger.fatal("Empty/null value for path to config.properties file was specified. This MUST be specified as \"-c /path/to/config.properties\" "
+						+ "or do not specify anything and the default will be src/main/resources/config.properties");
+			}
+//			if (noAuthConfigSpecified)
+//			{
+//				logger.fatal("Empty/null value for path to auth.properties file was specified. This MUST be specified as \"-c /path/to/auth.properties\" "
+//						+ "or do not specify anything and the default will be src/main/resources/auth.properties");
+//			}
+			throw new Error("Empty/null values were given for the path to a config file. The program cannot run withouth the config file.");
```

Comment:
I think Errors are usually reserved for something for the JVM.  Could a RuntimeException be used?

Reply:
No, anyone can use an Error, not just the JVM.

> An Error is a subclass of Throwable that indicates serious problems that a reasonable application should not try to catch. 

See: https://docs.oracle.com/javase/8/docs/api/java/lang/Error.html

Reply:
Yes, anyone can use Error but "Effective Java" (the accepted guide book for Java best practices) recommends using RuntimeException or a subclass.  I think it is the accepted convention.

Is there a reason Error is preferred over RuntimeException here?  They have the same semantic meaning and exceptions are usually preferred in program logic unless it's a system thing like an out of memory error.

Reply:
Because according to the Java spec, with a RuntimeException, "recovery may still be possible". (https://docs.oracle.com/javase/specs/jls/se7/html/jls-11.html#jls-11.1.1)

I'm using the `Error` in a place where recovery _is not possible_.

---

Effective Java recommends against _subclassing_ `Error`:
> ...it's best not to implement any new Error subclasses. Therefore, all of the unchecked throwables you implement should subclass RuntimeException.
[page 244].

Effective Java also says that RuntimeExceptions should indicate programming errors. I'm not sure this particular case is a programming error: Someone ran the program and did not specify the path to the configuration file. That seems like a user error, in the sense that the user ran the program with incorrect inputs.

Reply:
I think the context of "recovery is not possible" refers to an environmental problem like a memory issue as opposed to a problem with the program state (I think this is what is meant by "programming errors" - an issue to with the program and/or its input/state) where handling it is possible even if it's not desirable, as in this case.  I still think an exception may be better as it gives the caller the possibility of doing something with the issue where are error doesn't and it can be thrown up the stack to terminate the program or caught explicitly and the program terminated explicitly.

It's a bit of hair splitting in terms of Error vs Exception, so I'm okay if you want to keep this as is, but I do think Exception is better for handling program/state/logic related issues.